recommenders/datasets/mock/movielens.py

```diff
@@ -0,0 +1,140 @@
+    for more information.
+    """
+    # Some notebooks will do a cross join with userID and itemID,
+    # a sparse range for these IDs can slow down the notebook tests
+    userID: Series[int] = Field(in_range={"min_value": 1, "max_value": 10})
+    itemID: Series[int] = Field(in_range={"min_value": 1, "max_value": 10})
+    rating: Series[float] = Field(in_range={"min_value": 1, "max_value": 5})
+    timestamp: Series[str] = Field(eq="2022-2-22")
+    title: Series[str] = Field(eq="foo")
+    genre: Series[str] = Field(eq="genreA|0")
+
+    @classmethod
+    def get_df(
+        cls,
+        size: int = 3, seed: int = 100,
+        keep_first_n_cols: Optional[int] = None,
+        keep_title_col: bool = False, keep_genre_col: bool = False,
+    ) -> pandas.DataFrame:
+        """Return fake movielens dataset as a Pandas Dataframe with specified rows.
+
+        Args:
+            size (int): number of rows to generate
+            seed (int, optional): seeding the pseudo-number generation. Defaults to 100.
+            keep_first_n_cols (int, optional): keep the first n default movielens columns.
+            keep_title_col (bool): remove the title column if False. Defaults to True.
+            keep_genre_col (bool): remove the genre column if False. Defaults to True.
+
+        Returns:
+            pandas.DataFrame: a mock dataset
+        """
+        schema = cls.to_schema()
+        if keep_first_n_cols is not None:
+            if keep_first_n_cols < 1 or keep_first_n_cols > len(DEFAULT_HEADER):
+                raise ValueError(f"Invalid value for 'keep_first_n_cols': {keep_first_n_cols}. Valid range: [1-{len(DEFAULT_HEADER)}]")
+            schema = schema.remove_columns(DEFAULT_HEADER[keep_first_n_cols:])
+        if not keep_title_col:
+            schema = schema.remove_columns([DEFAULT_TITLE_COL])
+        if not keep_genre_col:
+            schema = schema.remove_columns([DEFAULT_GENRE_COL])
+
+        random.seed(seed)
+        # For more information on data synthesis, see https://pandera.readthedocs.io/en/latest/data_synthesis_strategies.html
+        return schema.example(size=size)
+
+    @classmethod
+    def get_spark_df(
+        cls,
+        spark,
+        size: int = 3, seed: int = 100,
+        keep_title_col: bool = False, keep_genre_col: bool = False,
+        tmp_path: Optional[str] = None,
+    ):
+        """Return fake movielens dataset as a Spark Dataframe with specified rows
+
+        Args:
+            spark (SparkSession): spark session to load the dataframe into
+            size (int): number of rows to generate
+            seed (int): seeding the pseudo-number generation. Defaults to 100.
+            keep_title_col (bool): remove the title column if False. Defaults to False.
+            keep_genre_col (bool): remove the genre column if False. Defaults to False.
+            tmp_path (str, optional): path to store files for serialization purpose
```

Comment:
does pandera return pure random data or it follows some kind of distribution?

@yueguoguo and Simon did a random generator based on scikit or similar? what is the benefit of pandera vs creating our own random data with a library that we already use in the repo like sklearn or numpy?

Reply:
> does pandera return pure random data or it follows some kind of distribution?

Under the hood, [pandera](https://pandera.readthedocs.io/en/latest/data_synthesis_strategies.html#data-synthesis-strategies) uses hypothesis, a popular data generation tool, to generate the fake data. By default, hypothesis uses the global pRNG in python to generate the random data, so we can provide deterministic random data with a fixed seed (it is what we are doing right now). For more detail [see this doc](https://hypothesis.readthedocs.io/en/latest/details.html#making-random-code-deterministic).

For the pros & cons of pandera vs implementation, I think there are a few arguments here:
1. Avoid re-inventing the wheel (as always 😝).
2. Easier use: data generation follows the validation rules you declared in the schema (for example, you can restrict the range of certain columns and random data will NOT provide anything outside the range).
3. Pandera's data schema is not used only for data generation, but also for [data validation](https://pandera.readthedocs.io/en/latest/schema_models.html). This is a bit biased anecdote, but data validation is such an important piece and a paint-point in a data pipeline, some of my previous work would have been longer to debug without neat validation like one provided by Pandera. This tool is very similar to [pydantic](https://pydantic-docs.helpmanual.io/) in data model validation but designed for `pandas.DataFrame` specially.

Reply:
got it, makes sense